sources/Map/Map.cpp

```diff
@@ -149,6 +150,7 @@ namespace BBM
 		scene->addEntity("Breakable Block")
 			.addComponent<PositionComponent>(coords)
 			.addComponent<HealthComponent>(1, &MapGenerator::wallDestroyed)
+			.addComponent<BonusComponent>()
```

Comment:
Your walls should not have that kind of component, to spawn bonuses use the HealthComponent's callback (wallDestroyed).